### ISSUE TITLE
Combine the hints and actions systems

### ISSUE THREAD

**Issue by Keavon** [GraphiteEditor/Graphite#741]
This lets us avoid redundantly specifying both the presently available actions (which are currently valid hotkeys) and curated hints (which are drawn for the user). By combining them, it forces us to be comprehensive about our hint list. We still need to provide the human-curated list of hints where a name and grouping is provided to render neatly for human consumption, but the attached actions would thus become part of the hints instead of being a separate system. Pulling off this change requires that we increase the capabilities of the input mapper for specifying how mouse interaction works with keys, and avoiding our current way of telling some messages to check for the status of certain modifier keys in the handler (those which use `refresh_keys` in the input mapper). We need to make changes to the system in order to be able to accurately derive the mouse and keyboard icons from the input mapper entry.

**Comment by TrueDoctor** [comment]
@Keavon  Is this still relevant?

**Comment by Keavon** [comment]
Yes, eventually. Next time I'm on a code cleanup binge I might even go for it.
